community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/xaframework/XaLogicalLog.java

```diff
@@ -1203,40 +1210,52 @@ public synchronized long rotate() throws IOException
         {
             long firstEntryPosition = getFirstStartEntry( endPosition );
             fileChannel.position( firstEntryPosition );
-            msgLog.logMessage( "Rotate log first start entry @ pos=" +
-                    firstEntryPosition + " out of " + xidIdentMap );
+            rotationMessage( currentVersion,
+                    "Rotate log first start entry @ pos=" + firstEntryPosition + " out of " + xidIdentMap );
         }
 
+        rotationMessage( currentVersion,
+                "Log header written. Starting migration of on-going transactions..." );
         LogBuffer newLogBuffer = instantiateCorrectWriteBuffer( newLog );
-        partialTransactionCopier.copy(/*from = */fileChannel, /* to= */newLogBuffer, /* targetLogVersion= */logVersion+1);
-
+        partialTransactionCopier.copy(
+                /*from = */fileChannel, /* to= */newLogBuffer, /* targetLogVersion= */logVersion+1);
         newLogBuffer.force();
         newLog.position( newLogBuffer.getFileChannelPosition() );
-        msgLog.logMessage( "Rotate: old log scanned, newLog @ pos=" +
-                newLog.position(), true );
         newLog.force( false );
+        rotationMessage( currentVersion,
+                "Old log scanned, on-going transactions copied and forced. New log position=" + newLog.position() +
+                ". Changing log version and setting new log as active..." );
+
         releaseCurrentLogFile();
         setActiveLog( newActiveLog );
-
         renameLogFileToRightVersion( currentLogFile, endPosition );
         xaTf.getAndSetNewVersion();
-
         this.logVersion = xaTf.getCurrentVersion();
         if ( xaTf.getCurrentVersion() != (currentVersion + 1) )
         {
             throw new IOException( "Version change failed, expected " + (currentVersion + 1) + ", but was " +
                     xaTf.getCurrentVersion() );
         }
+
+        rotationMessage( currentVersion, "New log set as active. Pruning archived log files..." );
         pruneStrategy.prune( this );
+        rotationMessage( currentVersion, "Archived log files pruned. Completing rotation..." );
+
         fileChannel = newLog;
         positionCache.putHeader( logVersion, lastTx );
         instantiateCorrectWriteBuffer();
-        msgLog.logMessage( "Log rotated, newLog @ pos=" +
-                writeBuffer.getFileChannelPosition() + ", version " + logVersion +
-                " and last tx " + previousLogLastCommittedTx, true );
+        rotationMessage( currentVersion,
+                "Log rotated completed. New log position=" + writeBuffer.getFileChannelPosition() +
+                ", version=" + logVersion + " and lastTx=" + previousLogLastCommittedTx + "." );
```

Comment:
"Log rotation completed" or "Log rotated".
version => newVersion ?

Not necessarily a problem, but the rotationMessage prefix message "Log Rotation [ x ]:" will have x as oldVersion early in this function and then newVersion here. Might be confusing, and since newVersion is already printed (version=y) then using oldVersion always for [ x ] might be better.